opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp

```diff
@@ -198,6 +209,11 @@ class SimulatorFullyImplicitBlackoilEbos
                                     prev_well_state,
                                     restorefilename,
                                     desiredRestoreStep );
+            initHydroCarbonState(state, phaseUsage_, Opm::UgGridHelpers::numCells(grid()), has_disgas_, has_vapoil_);
+            initHysteresisParams(state);
+            // communicate the restart solution to ebos
+            convertInput(0, state, ebosSimulator_);
```

Comment:
Not wrong. I just wanted to make it very clear that  the simulator
1) gets a state either from the initialization or one of the restart approaches
2) convert the state to the solution state in ebos
3) clear the cache in ebos.